### PR TITLE
Add pragmas completions

### PR DESCRIPTION
**Pull request by gdevanla**
This PR adds the completions of language pragmas to the `Pragmas` plugin.  Once this change is accepted, the completion code [here](https://github.com/haskell/ghcide/blob/baafe2cb82215f992ded3e3c1bd03507553aae86/src/Development/IDE/Plugin/Completions/Logic.hs#L564-L565) can be removed.

Also, starts addressing https://github.com/haskell/ghcide/issues/936

### CODE REVIEW

File: plugins/default/src/Ide/Plugin/Pragmas.hs [hunk 1]
```diff
@@ -160,3 +164,56 @@ possiblePragmas =
...
+                         preselect = Nothing
+                         sortText = Nothing
+                         filterText = Nothing
+                         insertText = Nothing
+                         insertTextFormat = Nothing
+                         textEdit = Nothing
+                         additionalTextEdits = Nothing
+                         commitCharacters = Nothing
+                         command = Nothing
+                         xd = Nothing
```
Comment (Ailrun) [inline plugins/default/src/Ide/Plugin/Pragmas.hs:218]: Would you mind if I ask you to change this to use record syntax? Looks a bit verbose to me.

File: stack.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-resolver: lts-14.27 # Last 8.6.5
+resolver: nightly-2020-11-22
```
Comment (Ailrun) [inline stack.yaml:1]: Could you explain why did you change the development resolver?

File: plugins/default/src/Ide/Plugin/Pragmas.hs (hunk 1 above)
Comment (gdevanla) [reply plugins/default/src/Ide/Plugin/Pragmas.hs:218]: yes. I tried it. I had to enable 'DuplicateRecords' and also had to move around imports to remove ambiguity. 

I will try again.

File: stack.yaml (hunk 2 above)
Comment (gdevanla) [reply stack.yaml]: I reverted back those changes. Checked in the wrong file.

**Comment by gdevanla** [comment]
BTW,  where do I add tests for this plugin?  I did not see any module. Should I create a new test module for `Pragmas` plugin?

**Comment by Ailrun** [comment]
@gdevanla `/test/functional/FunctionalCodeAction.hs` contains a `missingPragmaTests`. You can add a function similar to that to the module (with some test data in `/test/testdata` directory).

**Comment by gdevanla** [comment]
I feel like these tests should go to `/test/functional/Completion.hs`. But, lot of these tests are commented out with a `TODO: --TODO: Fix tests, some structural changed hav been made`. Any idea what needs to be done. I can do that as part of this change.

**Comment by Ailrun** [comment]
I think https://github.com/haskell/haskell-language-server/pull/657 has enabled the tests; could you rebase this PR on that PR? Then existing tests will give you some idea.

**Comment by gdevanla** [comment]
I have re-based and there are tests that already exercise this change.

**Review by Ailrun — APPROVED** [review]
Awesome, thank you for your contribution!

**Comment by jneira** [comment]
I am afraid that maybe we will have to mark the completion tests as broken, as ghcide-0.6 makes them fail with timeouts.
The pr worked against them so we will assume that will continue working after ghcode version bump (like the other completions).
